woocommerce/woocommerce-android
# [Hackweek] Customer more menu

Why
We want to match the iOS experience and include a customer section in the Hub Menu.
[pe5pgL-4L1-p2](https://href.li/?https://wp.me/pe5pgL-4L1)

### Description
This PR sets the ground for the customer's section in the hub menu. For displaying the customer list, I've refactored the customer's list from the order creation so I could forget about the common functionality (displaying customer, pagination, search, etc.) and implement only what is different (the customer selection logic)

### Testing information
TC1
1. Open the app
2. Tap o Menu
3. Check that now there is a Customers section
4. Tap on the Customers section
5. Check that the app displays the store's customers
6. Try the search bar and check that it is working as expected

TC2 (Optional)
1. Check that the select customer from the order creation flow keeps working as expected

⚠️ Tapping on a Customer will display a Toast message. Customer details will be handled in a different PR

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/bec8f7ad-bde7-4a3a-b184-e865534a0c62
